Bio/Seq.py

```diff
@@ -134,7 +149,10 @@ def __str__(self):
                 as_string = str(seq_obj)
 
         """
-        return self._data.decode("ASCII")
+        data = self._data
+        if not isinstance(data, (bytes, bytearray)):
```

Comment:
What else could it be?

Reply:
Any bytes-like object, meaning an object that returns `bytes` when calling `__getitem__` on it.
Currently in Biopython, only `TwoBitSequence` objects do that. This allows them to return subsequences without having to read the whole chromosome. I can imagine other uses for this in Biopython, for example for `UnknownSeq` and `DBSeq` in `BioSQL`.

Reply:
Interesting, and I start to see what you mean for ``UnknownSeq`` and ``DBSeq`` now.

I would suggest added a comment here, ``# e.g. the TwoBitSequence objects which can act like a view onto a sub-sequence``.

Having the 2bit C code start with some explanatory human readable text would with understanding the design. I had missed the significance of ``TwoBitSequence`` completely.